opsmngr/logs.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2019 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opsmngr
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	logsBasePath = "groups/%s/logCollectionJobs"
+)
+
+// LogsService is an interface for interfacing with the Log Collection Jobs
+// endpoints of the MongoDB Ops Manager API.
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/log-collection/
+type LogsService interface {
+	List(context.Context, string, *LogListOptions) (*Logs, *atlas.Response, error)
+	Get(context.Context, string, string, *LogListOptions) (*Log, *atlas.Response, error)
+	Retry(context.Context, string, string) (*atlas.Response, error)
+	Download(context.Context, string, string, io.Writer) (*atlas.Response, error)
+	Create(context.Context, string, *Log) (*Log, *atlas.Response, error)
+	Extend(context.Context, string, string, *Log) (*atlas.Response, error)
+	Delete(context.Context, string, string) (*atlas.Response, error)
+}
```

Comment:
This is not mandatory but up for discussion, what do you think of splitting this service in two?

one `LogCollectionService` and one `LogsService`

```suggestion
type LogCollectionService interface {
	List(context.Context, string, *LogListOptions) (*Logs, *atlas.Response, error)
	Get(context.Context, string, string, *LogListOptions) (*Log, *atlas.Response, error)
	Retry(context.Context, string, string) (*atlas.Response, error)	
	Create(context.Context, string, *Log) (*Log, *atlas.Response, error)
	Extend(context.Context, string, string, *Log) (*atlas.Response, error)
	Delete(context.Context, string, string) (*atlas.Response, error)
}

type LogsService interface {
	Download(context.Context, string, string, io.Writer) (*atlas.Response, error)
}
```

That way you don't need to create a new interface (good use of it btw) and gives us a clearer separation what what the service does